### PR TITLE
removes isflying helper proc

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

Removes the isflying helper proc.

## Why It's Good For The Game

It's so barely used and currently doesn't provide any real benefit. It's really inconsistent if isflying or movement_type&FLYING is used and there isn't really any reason why that should be the case.

This increases the overall readiblity of the code by removing a redudant proc.

## Changelog

not needed

### CODE REVIEW

File: code/datums/components/chasm.dm [hunk 1]
```diff
@@ -74,7 +74,7 @@
 			var/mob/buckled_to = M.buckled
 			if((!ismob(M.buckled) || (buckled_to.buckled != M)) && !droppable(M.buckled))
 				return FALSE
-		if(M.is_flying())
+		if(M.movement_type&FLYING)
```
Comment (ninjanomnom) [inline code/datums/components/chasm.dm]:
```suggestion
		if(M.movement_type & FLYING)
```

File: code/datums/components/slippery.dm [hunk 2]
```diff
@@ -15,5 +15,5 @@
 
 /datum/component/slippery/proc/Slip(datum/source, atom/movable/AM)
 	var/mob/victim = AM
-	if(istype(victim) && !victim.is_flying() && victim.slip(knockdown_time, parent, lube_flags, paralyze_time, force_drop_items) && callback)
+	if(istype(victim) && !victim.movement_type&FLYING && victim.slip(knockdown_time, parent, lube_flags, paralyze_time, force_drop_items) && callback)
```
Comment (ninjanomnom) [inline code/datums/components/slippery.dm]:
```suggestion
	if(istype(victim) && !victim.movement_type & FLYING && victim.slip(knockdown_time, parent, lube_flags, paralyze_time, force_drop_items) && callback)
```

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -768,7 +768,7 @@
 				throw_alert("gravity", /obj/screen/alert/highgravity)
 	else
 		throw_alert("gravity", /obj/screen/alert/weightless)
-	if(!override && !is_flying())
+	if(!override && !movement_type&FLYING)
```
Comment (ninjanomnom) [inline code/modules/mob/living/living.dm]:
```suggestion
	if(!override && !movement_type & FLYING)
```

File: code/modules/mob/living/living_movement.dm [hunk 4]
```diff
@@ -36,7 +36,7 @@
 	add_movespeed_modifier(MOVESPEED_ID_MOB_WALK_RUN_CONFIG_SPEED, TRUE, 100, override = TRUE, multiplicative_slowdown = mod)
 
 /mob/living/proc/update_turf_movespeed(turf/open/T)
-	if(isopenturf(T) && !is_flying())
+	if(isopenturf(T) && !movement_type&FLYING)
```
Comment (ninjanomnom) [inline code/modules/mob/living/living_movement.dm]:
```suggestion
	if(isopenturf(T) && !movement_type & FLYING)
```

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
you could have just made it into a define tbh lol

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
the `&` operations need to be parenthesized

**Pull request closed without merging**